open_city_profile/oidc.py

```diff
@@ -18,3 +27,67 @@ def authenticate(self, request, **kwargs):
             return None
         user, auth = user_auth_tuple
         return user
+
+
+class TunnistamoTokenExchange:
```

Comment:
I'm wondering if this is something that `django-helusers` can/should handle 🤔 [There's already some code](https://github.com/City-of-Helsinki/django-helusers/blob/3f57e79630302401765f4d5a566ca4e8361877e2/helusers/pipeline.py#L111-L140) that does something related, although definitely not exactly the same. But should this functionality be offered by `django-helusers`? Is this something that could benefit multiple projects?

Reply:
This comment is just playing with an idea. We can merge the implementation in this PR and consider extracting some functionality into `django-helusers` some other time.

Reply:
Indeed it looks like there is a function in `django-heluser` that accomplishes a similar thing, though it doesn exchange the authorization token, which is required here. But definitely the functionality in `TunnistamoTokenExchange` could be something that could be extracted to `django-helusers` at some point. :+1: